sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.integration.service.ts

```diff
@@ -25,32 +38,244 @@ export class IER12IntegrationService extends SFTPIntegrationBase<void> {
+      );
+      ierFileDetail.federalOverawardFlag = this.convertToYNFlag(
+        ierRecord.hasFederalOveraward,
+      );
+      ierFileDetail.restrictionFlag = this.convertToYNFlag(
+        ierRecord.hasRestriction,
+      );
+      ierFileDetail.scholasticStandingEffectiveDate =
+        this.getScholasticStandingEffectiveDate(
+          ierRecord.assessmentTriggerType,
+          ierRecord.scholasticStandingChangeType,
+          ierRecord.studyEndDate,
+        );
+      ierFileDetail.assessmentDate = ierRecord.assessmentDate;
+      ierFileDetail.withdrawalDate = this.getWithdrawalDate(
+        ierRecord.assessmentTriggerType,
+        ierRecord.scholasticStandingChangeType,
+        ierRecord.studyEndDate,
+      );
+      ierFileDetail.partnerFlag = this.convertToYNFlag(ierRecord.hasPartner);
+      ierFileDetail.parentalAssets = ierRecord.parentalAssets
+        ? combineDecimalPlaces(ierRecord.parentalAssets)
+        : 0;
+      ierFileDetail.coeStatus = ierRecord.coeStatus;
+      ierFileDetail.disbursementScheduleStatus =
+        ierRecord.disbursementScheduleStatus;
+      ierFileDetail.earliestDateOfDisbursement =
+        ierRecord.earliestDateOfDisbursement;
+      ierFileDetail.dateOfDisbursement = ierRecord.dateOfDisbursement;
+
+      ierFileDetail.disbursementCancelDate = ierRecord.disbursementCancelDate;
+      ierFileDetail.fundingDetails = this.getFundingDetails(
+        ierRecord.disbursementAwards,
+      );
       return ierFileDetail;
     });
     ierFileLines.push(...fileRecords);
     return ierFileLines;
   }
+
+  /**
+   * Identifies the given types of scholastic standing changes and
+   * returns the effective date which is study period end date of the
+   * offering which is created as a result of scholastic standing change.
+   * The change types `Student did not complete program` and `Student withdrew from program`
+   * are ignored.
+   * @param assessmentTriggerType assessment trigger type.
+   * @param scholasticStandingChangeType scholastic standing change type.
+   * @param studyEndDate study end date.
+   * @returns scholastic standing effective date.
+   */
+  private getScholasticStandingEffectiveDate(
+    assessmentTriggerType: AssessmentTriggerType,
+    scholasticStandingChangeType: StudentScholasticStandingChangeType,
+    studyEndDate: Date,
+  ): Date | null {
+    if (
+      assessmentTriggerType ===
+        AssessmentTriggerType.ScholasticStandingChange &&
+      (scholasticStandingChangeType ===
+        StudentScholasticStandingChangeType.StudentCompletedProgramEarly ||
+        scholasticStandingChangeType ===
+          StudentScholasticStandingChangeType.ChangeInIntensity)
+    ) {
+      return studyEndDate;
+    }
+    return null;
+  }
+
+  /**
+   * Get application status code based on application status.
+   * @param applicationStatus application status.
+   * @returns application status code.
+   */
+  private getApplicationStatusCode(
```

Comment:
I was wondering that maybe we could make use of Map in Typescript like in: https://stackoverflow.com/a/55123512/7983709

Reply:
Like this??

const applicationStatusMap = {
  [ApplicationStatus.Submitted]: ApplicationStatusCode.Submitted,
  [ApplicationStatus.InProgress]: ApplicationStatusCode.InProgress,
  [ApplicationStatus.Assessment]: ApplicationStatusCode.Assessment,
  [ApplicationStatus.Enrolment]: ApplicationStatusCode.Enrolment,
  [ApplicationStatus.Completed]: ApplicationStatusCode.Completed,
  [ApplicationStatus.Cancelled]: ApplicationStatusCode.Cancelled,
};


For calling
const statusCode = applicationStatusMap[applicationStatus];


I think we have not used Map widely in our project, nice one to try.